packages/ui/v2/core/layouts/SettingsLayout.tsx

```diff
@@ -1,99 +1,48 @@
-
-const tabs = [
-  {
-    name: "my_account",
-    href: "/settings/profile",
-    icon: Icon.FiUser,
-    children: [
-      { name: "profile", href: "/settings/my-account/profile" },
-      { name: "general", href: "/settings/my-account/general" },
-      { name: "calendars", href: "/settings/my-account/calendars" },
-      { name: "conferencing", href: "/settings/my-account/conferencing" },
-      { name: "appearance", href: "/settings/my-account/appearance" },
-      // TODO
-      { name: "referrals", href: "/settings/my-account/referrals" },
-    ],
-  },
-  {
-    name: "security",
-    href: "/settings/security",
-    icon: Icon.FiKey,
-    children: [
-      //
-      { name: "password", href: "/settings/security/password" },
-      { name: "2fa_auth", href: "/settings/security/two-factor-auth" },
-    ],
-  },
-  {
-    name: "billing",
-    href: "/settings/billing",
-    icon: Icon.FiCreditCard,
-    children: [
-      //
-      { name: "invoices", href: "/settings/billing" },
-    ],
-  },
-  {
-    name: "developer",
-    href: "/settings/developer",
-    icon: Icon.FiTerminal,
-    children: [
-      //
-      { name: "webhooks", href: "/settings/developer" },
-      { name: "api_keys", href: "/settings/developer" },
-      { name: "embeds", href: "/settings/developer" },
-    ],
-  },
-  {
-    name: "teams",
-    href: "/settings/teams",
-    icon: Icon.FiUsers,
-  },
-  {
-    name: "admin",
-    href: "/settings/admin",
-    icon: Icon.FiLock,
-    adminRequired: true,
-    children: [
-      //
-      { name: "impersonation", href: "/settings/admin/impersonation" },
-      { name: "apps", href: "/settings/admin/apps" },
-      { name: "users", href: "/settings/admin/users" },
-    ],
-  },
-];
+import MobileSettingsContainer from "../navigation/MobileSettingsContainer";
+import SettingsSidebarContainer from "../navigation/SettingsSidebarContainer";
 
 export default function SettingsLayout({
   children,
   ...rest
 }: { children: React.ReactNode } & ComponentProps<typeof Shell>) {
+  const [sideContainerOpen, setSideContainerOpen] = useState(false);
   return (
-    <Shell
-      flexChildrenContainer
-      {...rest}
-      SidebarContainer={
-        <VerticalTabs tabs={tabs} className="py-3 pl-3">
-          <VerticalTabItem
-            name="Settings"
-            href="/"
-            icon={Icon.FiArrowLeft}
-            textClassNames="text-md font-medium leading-none text-black"
-            className="mb-1"
-          />
-        </VerticalTabs>
-      }>
-      <div className="flex-1 [&>*]:flex-1">
-        <ShellHeader />
-        {children}
+    <Shell flexChildrenContainer {...rest} SidebarContainer={<SettingsSidebarContainer />}>
+      <div
+        className={classNames(
+          "absolute z-40 h-screen w-screen bg-black opacity-50",
+          sideContainerOpen ? "" : "hidden"
+        )}
+        onClick={() => {
+          setSideContainerOpen(false);
+        }}
+      />
```

Comment:
This div provides the dark filter on the screen when the side drawer is open on mobile view. Checked and it doesn't cause unnecessary rerendering